databricks_mcp/src/databricks_mcp/connector.py

```diff
@@ -0,0 +1,572 @@
+            f"Headers: {dict(resp.headers)}"
+        )
+
+    www_auth = resp.headers.get(WWW_AUTHENTICATE_HEADER)
+
+    # 1. Header method
+    if www_auth:
+        url_from_header = _try_resource_metadata_from_header(www_auth)
+        if url_from_header:
+            try:
+                metadata = _fetch_json(url_from_header, "Protected Resource metadata (from header)")
+                return metadata, www_auth
+            except RuntimeError as e:
+                logger.warning(f"Failed to fetch from header URL: {e}")
+                # fall through to well-known fallback
+
+    # 2. Well-known fallback
+    c1, c2 = _build_well_known_candidates(mcp_url)
+    for candidate in (c1, c2):
+        try:
+            metadata = _fetch_json(candidate, "Protected Resource metadata (well-known)")
+            return metadata, www_auth  # Return www_auth even if using well-known (may be None)
+        except RuntimeError as e:
+            logger.debug(f"Failed to fetch from {candidate}: {e}")
+            continue
+
+    raise RuntimeError(
+        "Could not discover Protected Resource Metadata via header or well-known URIs. "
+        "Ensure the MCP server supports OAuth discovery."
+    )
+
+
+# ---------------- AS Discovery ----------------
+
+
+def discover_authorization_server_metadata(resource_meta: Dict[str, Any]) -> Dict[str, Any]:
+    """Discover Authorization Server metadata from Protected Resource metadata.
+
+    Extracts the authorization server URL from the resource metadata and
+    attempts to fetch the authorization server metadata from well-known endpoints.
+
+    Args:
+        resource_meta: Protected Resource metadata dictionary
+
+    Returns:
+        Authorization Server metadata as a dictionary
+
+    Raises:
+        RuntimeError: If authorization server URL cannot be determined or metadata cannot be fetched
+    """
+    if not resource_meta:
+        raise ValueError("resource_meta cannot be empty")
+
+    logger.info("Discovering Authorization Server metadata")
+
+    # Extract authorization server base URL
+    if "authorization_servers" in resource_meta and resource_meta["authorization_servers"]:
+        as_base = resource_meta["authorization_servers"][0]
+        logger.debug(f"Using authorization_servers[0]: {as_base}")
+    elif "authorization_server" in resource_meta:
+        as_base = resource_meta["authorization_server"]
+        logger.debug(f"Using authorization_server: {as_base}")
+    elif "issuer" in resource_meta:
+        as_base = resource_meta["issuer"]
+        logger.debug(f"Using issuer: {as_base}")
+    else:
+        raise RuntimeError(
+            "Could not determine authorization server URL from Protected Resource metadata. "
+            "Expected one of: authorization_servers, authorization_server, or issuer. "
+            f"Available keys: {list(resource_meta.keys())}"
+        )
+
+    # Build candidate well-known URLs
+    candidates = [
```

Comment:
This makes sense for issues without path components, lets also add in the stuff for the path components specified here: https://modelcontextprotocol.io/specification/draft/basic/authorization#authorization-server-metadata-discovery
<img width="802" height="542" alt="image" src="https://github.com/user-attachments/assets/6dc425a5-5a26-46af-9c03-5130e6b25715" />